cosmwasm/ibc-union/core/light-client-interface/src/lib.rs

```diff
@@ -170,6 +186,71 @@ fn client_impl<T: IbcClient>(
     Ok(addr)
 }
 
+pub struct StateUpdate<T: IbcClient> {
+    pub height: u64,
+    pub client_state: Option<T::ClientState>,
+    pub consensus_state: T::ConsensusState,
+    pub storage_writes: StorageWrites,
+}
+
+impl<T: IbcClient> StateUpdate<T> {
+    pub fn new(height: u64, consensus_state: T::ConsensusState) -> Self {
+        StateUpdate {
+            height,
+            consensus_state,
+            client_state: None,
+            storage_writes: vec![],
+        }
+    }
+
+    pub fn set_client_state(mut self, client_state: T::ClientState) -> Self {
+        self.client_state = Some(client_state);
+        self
+    }
+
+    pub fn add_storage_write<V: Encode<T::Encoding>>(mut self, key: Bytes, value: V) -> Self {
+        self.storage_writes.push((key, value.encode().into()));
+        self
+    }
+}
+
+/// Client creation output type
+pub struct ClientCreation<T: IbcClient> {
+    /// The client state that is going to be stored by IBC. If set to `None`, IBC will store the
+    /// client state given by the creator as is
+    pub client_state: Option<T::ClientState>,
```

Comment:
minor api nit: i think it would be better if the client state must always be returned as it is to be stored. removes a logic branch and makes it simpler

Reply:
I initially do this to be able to do the following:
```
        Ok(ClientCreation::empty())
```
this simplifies the client codes where you absolutely dont need to care about this stuff